### PR TITLE
Recommendation service e2e testing

### PR DESCRIPTION
**Pull request by mic-max**
Towards #195 

Merge after: #196 
(If anyone knows a good git workflow for creating a PR based off another one that hasn't yet been merged yet please let me know so I can make these PRs cleaner since this one include all the previous branches commits - all I did was `git checkout -b recc-service-test` while on the `mic-max:e2e-tests` branch` anyways... shoot me a dm 😄)

## Changes

- Delete the `client.py`, essentially a manual version of this test
- Open port to localhost so we can connect to the container from the test program

The 1 test sends 5 product ids. this makes sure that the recommendation service always sends back the same array 4 product ids (random ordering) - it caps at 5 recommendations and wont include anything you sent in the request. I was originally going to test it just by including 1 id in the request but then its possible the service sends back 5 products that dont include the one in my request just by chance. So this way is more robust, still not perfect though since the random generator isn't seeded and more reasons I won't get into.

### CODE REVIEW

**Review by julianocosta89 — CHANGES_REQUESTED** [review]
I've tried the test and they failed in 4 out 5 tests, was that intended?

```shell
✖ payment: valid credit card Rejected promise returned by test
✖ payment: invalid credit card 
✖ payment: amex credit card not allowed 
✖ payment: expired credit card 
✔ recommendation: list products
```

File: test/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Service Testing
+
+Testing gRPC services as black boxes.
+
+1. Start the services you want to test with `docker compose up <service>`
```
[the remark this answers is not in the document]
Comment (mic-max) [reply test/README.md:5]: Ya if the service has changed it should be rebuilt. I can make it include the `--build` flag

**Comment by mic-max** [comment]
> I've tried the test and they failed in 4 out 5 tests, was that intended?
> 
> ```shell
> ✖ payment: valid credit card Rejected promise returned by test
> ✖ payment: invalid credit card 
> ✖ payment: amex credit card not allowed 
> ✖ payment: expired credit card 
> ✔ recommendation: list products
> ```

The payment service had an issue with it that I fixed in [this PR](https://github.com/open-telemetry/opentelemetry-demo/pull/197) which ill reduce to only the fix

**Review by cartersocha — APPROVED** [review]
LGTM

**Comment by julianocosta89** [comment]
Got the following now:
image

Are we expecting only `recommendation: list products` to be valid in this PR?
If yes, I'm good to approve.

**Comment by mic-max** [comment]
> Are we expecting only `recommendation: list products` to be valid in this PR? If yes, I'm good to approve.

Correct. The payment test failures were fixed in [this PR](https://github.com/open-telemetry/opentelemetry-demo/pull/230)

**Pull request merged**